ehsm_kms_service/quote_manager_apis.js

```diff
@@ -0,0 +1,260 @@
+            return
+        }
+        if (!checkStringParam(nonce, true, NONCE_LENGTH_MAX)) {
+            res.send(_result(400, `nonce cannot be empty, must be string and length not more than ${NONCE_LENGTH_MAX}.`))
+            return
+        }
+        if (!checkStringParam(policyId, true, UUID_LENGTH)) {
+            res.send(_result(400, `policyId cannot be empty, must be string and length equal ${UUID_LENGTH}.`))
+            return
+        }
+        if (!check_UUID_format(policyId)) {
+            res.send(_result(400, 'policyId format wrong.'))
+            return
+        }
+
+        // query quote policy by policyId
+        const query_quote_policy = {
+            selector: {
+                appid,
+                policyId
+            },
+            fields: ['mrenclave', 'mrsigner'],
+            limit: 1
+        }
+        let quote_policy_res = await DB.partitionedFind('quote_policy', query_quote_policy)
+        if (quote_policy_res.docs.length != 1) {
+            res.send(_result(400, 'Invalid  policyId.'))
+            return
+        }
+        mrenclave = quote_policy_res.docs[0].mrenclave
+        mrsigner = quote_policy_res.docs[0].mrsigner
+        napi_res = napi_result(action, res, [quote, mrsigner, mrenclave, nonce])
+        if (napi_res) {
+            let { error, hmac } = await gen_hmac(DB, appid, napi_res.result)
+            if (hmac.length > 0) {
+                napi_res.result.sign = hmac
+                res.send(napi_res)
+                return
+            } else {
+                res.send(_result(400, 'Internal error', {}))
+                return
+            }
+        }
+    } catch (e) {
+        console.info('verifyQuote :: ', e)
+        res.send(_result(500, 'Server internal error, please contact the administrator.'))
+        return
+    }
+}
+
+/**
+ * The UploadQuotePolicy Support uploading MRenclave and MRsigner and returning new policyid.
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controller
+ * @param {Object} payload
+ *          ==> {r}mrenclave(String [1~1024]) : a base64 string. eg. 'NTU4OTI1NTI1NTAwMDAwMDAwMDAw****'
+ *          ==> {r}mrsigner(String [1~1024]) : a base64 string. eg. 'NzAwMDAwMDAyMzEwMDAwMDAwMTM1M****'
+ * @returns {Object}
+ *          ==> {r}policyId(String) : a new policy ID. eg. '15ca0dd5-2d34-4221-a708-3171ffe6****'
+ */
+const uploadQuotePolicy = async (res, appid, payload, DB) => {
+    try {
+        const { mrenclave, mrsigner } = payload
+        if (!checkStringParam(mrenclave, true, QUOTE_LENGTH_MAX)) {
+            res.send(_result(400, `mrenclave cannot be empty, must be string and length not more than ${QUOTE_LENGTH_MAX}`))
+            return
+        }
+        if (!checkStringParam(mrsigner, true, NONCE_LENGTH_MAX)) {
+            res.send(_result(400, `mrsigner cannot be empty, must be string and length not more than ${NONCE_LENGTH_MAX}`))
```

Comment:
you must also be notice that use the same format in the log print.

Reply:
i mean the mr_signer/mr_enclave

Reply:
all mrsigner and mrenclave will be change to mr_signer/mr_enclave.